File: src/main/java/com/iceprosurface/deathresurrection/DeadListener.java
package com.iceprosurface.deathresurrection;

import jdk.vm.ci.meta.Local;
import org.bukkit.*;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;
import org.bukkit.event.player.PlayerRespawnEvent;

import java.util.List;

public class DeadListener implements Listener {
    @EventHandler
    public void onDeath(PlayerDeathEvent playerDeathEvent) {
        Player player = playerDeathEvent.getEntity();
        if (player instanceof Player) {
            DeathResurrection instance = DeathResurrection.getInstance();
            if (!instance.isPlayerBaned(player)) {
                player.getServer().broadcastMessage(ChatColor.YELLOW + player.getDisplayName() + "已经在极限模式中死亡，现在将其切换为英灵模式");
                playThunder(player);
            } else {
                player.getServer().broadcastMessage("英灵" + player.getDisplayName() + "再一次阵亡了");
            }
            DeathResurrection.getInstance().banPlayer(player);
        }
    }

    private void playThunder(Player player) {
        DeathResurrection instance = DeathResurrection.getInstance();
        List<Integer> respawnPoint = instance.config.getConfig().getIntegerList("RespawnPoint");
        String respawnWorld = instance.config.getConfig().getString("RespawnWorld");
        if (respawnWorld == null) {
            return;
        }
        World world = instance.getServer().getWorld(respawnWorld);
        if (world == null) {
            return;
        }
        Location respawn = new Location(
                world,
                respawnPoint.get(0).doubleValue(),
                respawnPoint.get(1).doubleValue(),
                respawnPoint.get(2).doubleValue()
        );
        Sound sound = Sound.ITEM_TRIDENT_THUNDER;
        world.playSound(respawn, sound, 3.0F, 0.5F);
        for (Player worldPlayer : world.getPlayers()) {
            if (player == worldPlayer) {
                continue;
            }
            worldPlayer.playSound(worldPlayer.getLocation(), sound,  2f, 0.3f);
        }
    }

}

File: src/main/kotlin/com/iceprosurface/deathresurrection/DeadListener.kt
package com.iceprosurface.deathresurrection

import org.bukkit.ChatColor
import org.bukkit.Location
import org.bukkit.Sound
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

class DeadListener : Listener {
    @EventHandler
    fun onDeath(playerDeathEvent: PlayerDeathEvent) {
        val player = playerDeathEvent.entity
        val instance: DeathResurrection = DeathResurrection.instance
        if (!instance.isPlayerBaned(player)) {
            player.server.broadcastMessage(ChatColor.YELLOW.toString() + player.displayName + "已经在极限模式中死亡，现在将其切换为英灵模式")
            playThunder(player)
        } else {
            player.server.broadcastMessage("英灵" + player.displayName + "再一次阵亡了")
        }
        DeathResurrection.instance.banPlayer(player)
    }

    private fun playThunder(player: Player) {
        val instance: DeathResurrection = DeathResurrection.instance
        val respawnPoint = instance.config.config?.getIntegerList("RespawnPoint")
        val respawnWorld = instance.config.config?.getString("RespawnWorld") ?: return
        val world = instance.server.getWorld(respawnWorld) ?: return
        if (respawnPoint == null || respawnPoint.size != 3) {
            return
        }
        val respawn = Location(
                world,
                respawnPoint[0].toDouble(),
                respawnPoint[1].toDouble(),
                respawnPoint[2].toDouble()
        )
        val sound = Sound.ITEM_TRIDENT_THUNDER
        world.playSound(respawn, sound, 3.0f, 0.5f)
        for (worldPlayer in world.players) {
            if (player === worldPlayer) {
                continue
            }
            worldPlayer.playSound(worldPlayer.location, sound, 2f, 0.3f)
        }
    }
}